src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AlignmentRegion.java

```diff
@@ -14,76 +14,92 @@
         this.assemblyId = assemblyId;
+        this.contigId = contigId;
+
         this.forwardStrand = alnRgn.getStrand() == '+';
         final Cigar alignmentCigar = TextCigarCodec.decode(alnRgn.getCigar());
         this.forwardStrandCigar = forwardStrand ? alignmentCigar : CigarUtils.invertCigar(alignmentCigar);
         this.referenceInterval = new SimpleInterval(alnRgn.getChrom(), (int) alnRgn.getPos() + 1, (int) (alnRgn.getPos() + forwardStrandCigar.getReferenceLength()));
         this.mapQual = alnRgn.getMQual();
+        this.mismatches = alnRgn.getNm();
+
         this.assembledContigLength = forwardStrandCigar.getReadLength() + SVVariantCallerUtils.getTotalHardClipping(forwardStrandCigar);
         this.startInAssembledContig = startOfAlignmentInContig();
         this.endInAssembledContig = endOfAlignmentInContig();
-        this.mismatches = alnRgn.getNm();
     }
 
-    public AlignmentRegion(final String assemblyId, final String contigId, final Cigar forwardStrandCigar, final boolean forwardStrand, final SimpleInterval referenceInterval, final int mapQual, final int startInAssembledContig, final int endInAssembledContig, final int mismatches) {
-        this.contigId = contigId;
+    @VisibleForTesting
+    public AlignmentRegion(final String assemblyId, final String contigId, final SimpleInterval referenceInterval,
+                           final Cigar forwardStrandCigar, final boolean forwardStrand, final int mapQual, final int mismatches,
+                           final int startInAssembledContig, final int endInAssembledContig) {
         this.assemblyId = assemblyId;
+        this.contigId = contigId;
+        this.referenceInterval = referenceInterval;
         this.forwardStrandCigar = forwardStrandCigar;
         this.forwardStrand = forwardStrand;
-        this.referenceInterval = referenceInterval;
         this.mapQual = mapQual;
+        this.mismatches = mismatches;
+        this.assembledContigLength = forwardStrandCigar.getReadLength() + SVVariantCallerUtils.getTotalHardClipping(forwardStrandCigar);
         this.startInAssembledContig = startInAssembledContig;
         this.endInAssembledContig = endInAssembledContig;
-        this.assembledContigLength = forwardStrandCigar.getReadLength() + SVVariantCallerUtils.getTotalHardClipping(forwardStrandCigar);
-        this.mismatches = mismatches;
     }
 
     public AlignmentRegion(final GATKRead read) {
-        this.assemblyId = null;
+        this.assemblyId = DUMMY_ASM_ID;
         this.contigId = read.getName();
+        this.referenceInterval = new SimpleInterval(read);
         this.forwardStrand = ! read.isReverseStrand();
         this.forwardStrandCigar = forwardStrand ? read.getCigar() : CigarUtils.invertCigar(read.getCigar());
-        this.referenceInterval = new SimpleInterval(read);
+        this.mapQual = read.getMappingQuality();
+        this.mismatches = read.hasAttribute("NM") ? read.getAttributeAsInteger("NM") : 0;
```

Comment:
Right, it was something I forgot to change. Will go with the special-value case.